pkg/openstack/common.go

```diff
@@ -646,6 +669,27 @@ func (ed *EndpointDetail) CreateRoute(
 			return ctrlResult, nil
 		}
 
+		// Delete the issued certificate if it exists and custom cert secret or direct TLS data was provided
+		if ed.Route.TLS.SecretName != nil || (ed.Route.OverrideSpec.Spec != nil && ed.Route.OverrideSpec.Spec.TLS != nil &&
+			(ed.Route.OverrideSpec.Spec.TLS.CACertificate != "" &&
+				ed.Route.OverrideSpec.Spec.TLS.Certificate != "" &&
+				ed.Route.OverrideSpec.Spec.TLS.Key != "")) {
```

Comment:
lets create a small func for the 2nd part to check the cert info was provided via the overrideSpec. makes it probably easier to understand